### PR TITLE
Features/add restart exit confirm

### PR DESCRIPTION
**Pull request by Bluebugs**

### CODE REVIEW

File: exit.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package fyneselfupdate
+
+import (
+	"log"
+
+	"fyne.io/fyne/v2"
+	"fyne.io/fyne/v2/dialog"
+)
+
+// NewExitCallback properly exit a Fyne App. If a window is specified, it will Hide
```
Comment (andydotxyz) [inline exit.go]: Why will it hide the window first?
Adding show seems more complexity too, why do this?
I understand that someone may want `win` to be optional, but I don't think a callback like this should be handling window lifecycle.

File: confirm.go [hunk 2]
```diff
@@ -41,3 +41,37 @@ func NewConfirmCallbackWithTimeout(win fyne.Window, timeout time.Duration) func(
...
+		if timeout > 0 {
+			ctx, fn := context.WithTimeout(context.Background(), timeout)
+			cancel = fn
+			go func() {
+				<-ctx.Done()
+				d.Hide()
+				resp <- defaultRestart
+			}()
+		}
+		d = dialog.NewConfirm("Application Update", "Do you wish to restart this application?\n", func(ok bool) {
```
Comment (andydotxyz) [inline confirm.go]: We might need to give the user context here - they did not request the restart directly.
Something that mentions that the update is ready and the application needs to be restarted, is this OK.

File: confirm.go [hunk 3]
```diff
@@ -41,3 +41,37 @@ func NewConfirmCallbackWithTimeout(win fyne.Window, timeout time.Duration) func(
...
+
+// NewRestartConfirmCallback create a GUI based confirmation to approve restarting the application after being updated
+func NewRestartConfirmCallback(win fyne.Window) func() bool {
+	return NewRestartConfirmCallbackWithTimeout(win, false, 0)
+}
+
+// NewRestartConfirmCallbackWithTimeout creates a GUI based restarting confirmation callback
+// with a timeout, after which time the question will be confirmed if defaultRestart is true.
+// This can assist in a "default to yes" update where computer may be unattended.
+func NewRestartConfirmCallbackWithTimeout(win fyne.Window, defaultRestart bool, timeout time.Duration) func() bool {
```
Comment (andydotxyz) [inline confirm.go]: This doesn't match the other `WithTimeout` which asserts that a non-zero timeout creates a default-to-yes situation.
timing out with no action seems to provide no benefit

**Review by andydotxyz — CHANGES_REQUESTED** [review]

File: exit.go (hunk 1 above)
Comment (Bluebugs) [reply exit.go]: There is potential strange behavior without. The application might start faster than this one close, leaving two windows visible for a short period of time which I think need to be handle by the default behavior.

I think the show was there due to some earlier test where I was hiding the windows straight after confirmation and I needed to handle the error case. I will check if that still work like expected.

Comment (andydotxyz) [reply exit.go]: Is it worth just quitting for now, seeing how it works in the wild, and potentially refining later if required?

Comment (Bluebugs) [reply exit.go]: I checked again, and indeed the show is a left over from a previous experience where I moved the win.Hide() inside the restart confirm callback. I had the order wrong in that function when I did the experiment. I think it is better to move back to Hide inside the restart confirm callback actually.

Comment (andydotxyz) [reply exit.go]: Cool. Let's shift the hide/show out for now

**Review by andydotxyz — APPROVED** [review]
Thanks, looks good

**Pull request merged**